lib/device_api/android/adb.rb

```diff
@@ -233,6 +233,34 @@ def self.screencap( serial, args )
         shell(serial, cmd)
       end
 
+      def self.check_ip_address(ipaddressandport)
+        unless ipaddressandport =~ /\A(\d{1,3})\.(\d{1,3})\.(\d{1,3})\.(\d{1,3}):[0-9]+\Z/ 
+             raise ADBCommandError.new("Invalid IP address and port " + ipaddressandport)
+        end
+      end
+
+      def self.connect(ipaddressandport)
+        check_ip_address(ipaddressandport)
+        cmd = "adb connect " + ipaddressandport
+        result = execute(cmd)
+        if result.stdout.to_s =~ /.*already connected to.*/
+          raise DeviceAlreadyConnectedError.new("Device " + ipaddressandport + " already connected")
```

Comment:
Embed in the string:
"Device #{ipaddressandport} already connected"
(Handles nil and unexpected values better)